server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java

```diff
@@ -347,4 +392,51 @@ private static ClusterState update(ClusterState currentState, PersistentTasksCus
             return currentState;
         }
     }
+
+    /**
+     * Class to periodically try to reassign unassigned persistent tasks.
+     */
+    private class PeriodicRechecker implements Runnable {
```

Comment:
I think it's time to extract a utility class that's shared with `org.elasticsearch.index.IndexService.BaseAsyncTask`